pkg/operator/resource/resourceapply/monitoring.go

```diff
@@ -102,70 +99,42 @@ func ApplyUnstructuredResourceImproved(
 		return existing, false, nil
 	}
 
-	// Ensure metadata field is present on the object.
 	existingCopy := existing.DeepCopy()
-	existingObjectMeta, found, err := unstructured.NestedMap(existingCopy.Object, "metadata")
-	if err != nil {
-		return nil, false, err
-	}
-	if !found {
-		return nil, false, errorsstdlib.New(fmt.Sprintf("metadata not found in the existing object: %s/%s", existing.GetNamespace(), existingCopy.GetName()))
-	}
-	requiredObjectMeta, found, err := unstructured.NestedMap(required.Object, "metadata")
-	if err != nil {
-		return nil, false, err
-	}
-	if !found {
-		return nil, false, errorsstdlib.New(fmt.Sprintf("metadata not found in the required object: %s/%s", required.GetNamespace(), required.GetName()))
-	}
 
-	// Cast the metadata to the correct type.
-	var existingObjectMetaTyped, requiredObjectMetaTyped metav1.ObjectMeta
-	err = runtime.DefaultUnstructuredConverter.FromUnstructured(existingObjectMeta, &existingObjectMetaTyped)
-	if err != nil {
-		return nil, false, err
-	}
-	err = runtime.DefaultUnstructuredConverter.FromUnstructured(requiredObjectMeta, &requiredObjectMetaTyped)
+	// Replace and/or merge certain metadata fields.
+	didMetadataModify := ptr.To(false)
```

Comment:
could you change to a non-pointer variable to avoid dereferencing bellow ? (`didMetadataModify := false`)